build/MPack.targets

```diff
@@ -0,0 +1,64 @@
+        b. Addin assembly attributes for metadata 
+      3. All language service binaries
+    -->
+
+    <PropertyGroup>
+      <MSBuildArtifactsDir>$(ArtifactsDir)msbuild\</MSBuildArtifactsDir>
+      <MPackSourcesDir>$(MSBuildArtifactsDir)sources\</MPackSourcesDir>
+      <AddinOutputPath>$(AddinDirectory)bin\$(Configuration)\net461\</AddinOutputPath>
+      <LanguageServiceName>Microsoft.MonoDevelop.LanguageServices.Razor</LanguageServiceName>
+      <LanguageServiceOutputPath>$(RepositoryRoot)src\$(LanguageServiceName)\bin\$(Configuration)\net46\</LanguageServiceOutputPath>
+      <MPackName>$(AddinName)_$(AddinVersion)</MPackName>
+      <MPackFileName>$(MPackName).mpack</MPackFileName>
+      <MPackOutputPath>$(BuildDir)$(MPackFileName)</MPackOutputPath>
+      <MPackZipFile>$(BuildDir)$(MPackName).zip</MPackZipFile>
+      <MPackManifest>$(AddinDirectory)Properties\_Manifest.addin.xml</MPackManifest>
+      <AddinInfoFilePath>$(MPackSourcesDir)addin.info</AddinInfoFilePath>
+    </PropertyGroup>
+    
+    <MakeDir Directories="$(MPackSourcesDir)" Condition="!Exists('$(MPackSourcesDir)')" />
+
+    <!-- We need to resolve the language service assemblies to generate an addin.info for the mpack -->
+    <XmlPeek XmlInputPath="$(MPackManifest)" Query="/ExtensionModel/Runtime/Import/@assembly">
+      <Output TaskParameter="Result" ItemName="LanguageServiceAssemblies" />
+    </XmlPeek>
+
+    <ItemGroup>
+      <AddinInfoLines Include="&lt;Addin id=&quot;$(AddinId)&quot; namespace=&quot;$(AddinNamespace)&quot; version=&quot;$(AddinVersion)&quot; name=&quot;$(AddinDetailedName)&quot; author=&quot;$(Authors)&quot; description=&quot;$(Description)&quot; category=&quot;$(AddinCategory)&quot;&gt;" />
+      <AddinInfoLines Include="  &lt;Runtime&gt;" />
+      <AddinInfoLines Include="    &lt;Import assembly=&quot;%(LanguageServiceAssemblies.Identity)&quot; /&gt;" />
+      <AddinInfoLines Include="    &lt;Import assembly=&quot;$(AddinName).dll&quot; /&gt;" />
+      <AddinInfoLines Include="  &lt;/Runtime&gt;" />
+      <AddinInfoLines Include="  &lt;Dependencies&gt;" />
+      <AddinInfoLines Include="  &lt;/Dependencies&gt;" />
+      <AddinInfoLines Include="&lt;/Addin&gt;" />
+    </ItemGroup>
+
+    <!-- Generate the addin.info and gather sources for mpack zipping-->
+    <WriteLinesToFile File="$(AddinInfoFilePath)" Lines="@(AddinInfoLines)" Overwrite="true" />
+    <Copy SourceFiles="$(LanguageServiceOutputPath)\%(LanguageServiceAssemblies.Identity)" DestinationFolder="$(MPackSourcesDir)" />
+    <Copy SourceFiles="$(AddinOutputPath)$(AddinName).dll" DestinationFolder="$(MPackSourcesDir)" />
+
+    <!-- We cannot use the ZipArchive task due to how it functions in CoreCLR. The archive it generates is unreadable by MonoDevelop. -->
+    <Exec Command="powershell.exe -NonInteractive -command &quot;&amp; { Compress-Archive -Path $(MPackSourcesDir)* -DestinationPath $(MPackZipFile) -Force; } &quot;" />
```

Comment:
You could add a custom build task to the repo and do all of this work in there. Might be cleaner than rendering xml and shelling out to powershell. 

https://github.com/aspnet/Coherence/blob/dev/build/tasks/ has an example of what it takes to write a task.

Reply:
Arent MSBuild tasks executed in a CoreCLR context?

Reply:
Yup. I thought you had an issue with the specific structure of the produced zip rather than the format (the format is pretty standard, no?)

Reply:
The CoreCLR implementation of ZipArchive was re-written and has had a significant number of bugs already logged/fixed (heck our own ZIpArchive task works around one of the bugs). I spent some time investigating what could be wrong but didn't get far.

Reply:
I think we should pursue whatever issues are blocking us in parallel to getting this work done. 

We want this to be able to build everywhere, the hack with powershell is a speed bump we have to cross.